lua/source/snippet.lua

```diff
@@ -51,13 +51,39 @@ local getNeosnippetItems = function(prefix)
   return complete_items
 end
 
+local getVsnipItems = function(prefix)
+  if vim.fn.exists("*vsnip#source#find") == 0 then return {} end
+  local snippetsList = api.nvim_call_function('vsnip#source#find', {api.nvim_buf_get_option(0, 'filetype')})
+  local complete_items = {}
+  if vim.tbl_isempty(snippetsList) == 0 then
+    return {}
+  end
+  local priority = vim.g.completion_items_priority['Vsnip']
+  for sourceIdx, source in pairs(snippetsList) do
+    for snippetIdx, snippet in pairs(source) do
+      for prefixIdx, word in pairs(snippet.prefix) do
+        local user_data = {hover = snippet.description}
+        local item = {}
+        item.word = word
+        item.kind = 'Vsnip'
+        item.priority = priority
+        item.user_data = user_data
+        match.matching(complete_items, prefix, item)
+      end
```

Comment:
Your work still needed but complete_items creation is bundled in `vim-vsnip` side
so we can use `vsnip#get_complete_items(bufnr)` instead of this codes.

Reply:
Hmm I personally thinks the current solution is better because it respect the priority and and score from fuzzy matching(it'll affect the order in the popup menu).

Reply:
OK, I will follow your opinion.

I recommend using `prefix[0]` instead of `for prefixIdx, word in pairs(snippet.prefix) do`.